src/nwb2bids/_converters/_dataset_converter.py

```diff
@@ -264,6 +265,20 @@ def write_participants_metadata(self) -> None:
         # Deduplicate all rows of the frame
         deduplicated_data_frame = full_participants_data_frame.drop_duplicates(ignore_index=True)
 
+        # Apply sanitization
+        deduplicated_data_frame["participant_id"] = deduplicated_data_frame["participant_id"].apply(
+            lambda participant_id: (
+                sanitize_participant_id(
+                    participant_id=participant_id,
+                    sanitization_level=self.run_config.sanitization_level,
+                    sanitization_file_path=self.run_config.sanitization_file_path,
+                    sanitization_report_context="DatasetConverter.write_participants_metadata",
+                )
+                if participant_id is not None
```

Comment:
```suggestion
                if pandas.notna(participant_id)
```

`participant_id is not None` is always `True` since for every Python `None` value loaded into a data frame it becomes `pandas.NA` upon retrieval, and `pandas.NA` is none `None`.

See the demo script at https://github.com/con/nwb2bids/pull/167/files#r2590276380 if want a demo.